pkg/cri/sbserver/sandbox_run.go

```diff
@@ -144,8 +143,17 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 		}
 	}()
 
+	controller, err := c.getSandboxController(sandbox.Config, sandbox.RuntimeHandler)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sandbox controller: %w", err)
+	}
+	platform, err := controller.Platform(ctx, sandbox.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query sandbox platform: %w", err)
+	}
+
 	userNsEnabled := false
-	if goruntime.GOOS != "windows" {
+	if platform.OS == "linux" {
```

Comment:
let's validate that only linux is valid here.. https://github.com/containerd/containerd/blob/main/platforms/database.go#L34

Reply:
@mikebrow as far as I understand, yes, Linux is the only valid here.

We use only these platforms and we only call `buildLinuxSpec()` if the OS is `== "linux"`:https://github.com/containerd/containerd/blob/40f26543bdc27cbe8b058ac082e91c5832bb1c41/pkg/cri/sbserver/container_create.go#L528-L539


Windows does not support userns and idmap mounts and, AFAIK darwin either. Or is there some way in which darwin can support this?

Reply:
> Or is there some way in which darwin can support this?

Nope. Darwin itself nope, Darwin virtualizing linux obviously yes 😆

Reply:
@dcantah right, but the platform in that case would be "linux" (and the GOOS will be darwin). So this code will work fine, right? Is this comment solved, then? :)

Reply:
Right, I think what's here is correct. None of this makes sense on anything but Linux, and obviously the point of this change is so that a sandbox runtime that emulates/virtualizes linux can benefit from this check now.

Reply:
Cool, that is what I thought. Solved, then! :)